code/script.js

```diff
@@ -1,325 +1,539 @@
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        `Yes, the person wears ${value}! Keep all people that wear ${value}`
       )
+      // Keep only the persons who has accessories value match with the current question
+      // For example: selected accessory is earing => keep only the ones who have earing
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `No, the person doesn't wear ${value}! Remove all people that wear ${value}`
       )
+
+       // Keep only the persons who doesn't have the accessory
+      // For example: selected accessory is hat => keep only the ones who do not wear hat
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
-    }
-  }
+    
+   } else if ( category === "job") {
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+      if (keep) {
+        alert(
+          `Yes, the person is a ${value}! Keep all people that are ${value}`
+        )
+        charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+      } 
+      else {
+        alert(
+          `No, the person isn't a ${value}! Remove all people that are ${value}`
+        )
+        //It will remove all persons who have the same value, and keep only the ones who don't
+        charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+      }
+    } 
+    else if (category === "hair" || category === "eyes") {
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+      if(keep) {
+        alert (`Yes, the person has ${value}! Keep all people that have ${value}`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+      } 
+      else {
+        alert(`No, the person doesn't has ${value}! Remove all people that have ${value}`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      }
+
```

Comment:
This is just preference but I like to tighten up the if/else-statements into fewer lines, for me it gives me a better overview. Maybe increase the indentation so it's easier to separate the main if/elses from the nested ones.